osu.Game/Graphics/Containers/HoldToConfirmContainer.cs

```diff
@@ -40,9 +40,13 @@ public abstract class HoldToConfirmContainer : Container
 
         private Bindable<double> holdActivationDelay;
 
-        [BackgroundDependencyLoader]
-        private void load(OsuConfigManager config)
+        [Resolved]
+        private OsuConfigManager config { get; set; }
+
+        protected override void LoadComplete()
         {
+            base.LoadComplete();
```

Comment:
Yeah this is a bit awkward... I presume this change was made so that the binding happens after BDL, because the override that makes this work in multiplayer relies on accessing a resolved `Player`?

I'd say that this is another strong argument to remove the binding logic from here. On the other side I can see doing so introducing duplication all over, but maybe it's fine because the logic is rather simple...?

Reply:
I've refactored to avoid this flow altogether, by baking the `dangerous` concept into the base class. I think this simplifies things a lot.